source/read.txt

```diff
@@ -30,5 +30,136 @@ Read Data from MongoDB
    /read/count
    /read/change-streams
 
+Overview
+--------
 
+On this page, you can see copyable code examples that show common methods you
+can use to read data from MongoDB by using the {+driver-short+}.
 
+.. tip::
+
+  To learn more about any of the methods shown on this page, see the link
+  provided in each section.
+
+To use an example from this page, copy the code example into the
+:ref:`sample application <scala-read-sample>` or your own application.
+Be sure to replace all placeholders in the code examples, such as ``<connection string URI>``, with
+the relevant values for your MongoDB deployment.
+
+.. _scala-read-sample:
+
+Sample Application
+~~~~~~~~~~~~~~~~~~
+
+.. include:: /includes/usage-examples/sample-app-intro.rst
+
+.. literalinclude:: /includes/usage-examples/SampleReadApp.scala
+   :language: scala
+   :copyable:
+   :linenos:
+   :emphasize-lines: 14-16
+
+Find One
+--------
+
+The following example retrieves a document that matches the criteria specified by the
+given filter:
+
+.. literalinclude:: /includes/usage-examples/ReadCodeExamples.scala
+   :start-after: start-find-one
+   :end-before: end-find-one
+   :language: scala
+   :copyable:
+   :dedent:
+
+To learn more about the ``find().first()`` method, see the :ref:`Retrieve Data <scala-retrieve>` guide.
+
+Find Multiple
+-------------
+
+The following example retrieves all documents that match the criteria specified by the
+given filter:
+
+.. literalinclude:: /includes/usage-examples/ReadCodeExamples.scala
+   :start-after: start-find-multiple
+   :end-before: end-find-multiple
+   :language: scala
+   :copyable:
+   :dedent:
+
+To learn more about the ``find()`` method, see the :ref:`Retrieve Data <scala-retrieve>` guide.
+
+Count Documents in a Collection
+-------------------------------
+
+The following example returns the number of documents in the specified collection:
+
+.. literalinclude:: /includes/usage-examples/ReadCodeExamples.scala
+   :start-after: start-count-collection
+   :end-before: end-count-collection
+   :language: scala
+   :copyable:
+   :dedent:
+
+To learn more about the ``countDocuments()`` method, see the :ref:`Count Documents <scala-count>` guide.
+
+Count Documents Returned from a Query
+-------------------------------------
+
+The following example returns the number of documents in the specified
+collection that match the criteria specified by
+the given filter:
```

Comment:
S: just to avoid repeating "specified" and make this more concise, you can just say "query criteria"
```suggestion
collection that match the query criteria:
```